orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/pipeline/MonitorCanaryStage.groovy

```diff
@@ -74,16 +73,19 @@ class MonitorCanaryStage implements StageDefinitionBuilder, CancellableStage {
       log.error("Unable to cancel canary '${canaryId}' in mine", e)
     }
 
-    def canaryStages = stageNavigator.ancestors(stage).findAll {
-      it.stageBuilder instanceof CanaryStage
+    Stage canaryStageInstance = stage.ancestors().find {
+      it.type == CanaryStage.PIPELINE_CONFIG_TYPE
     }
 
-    if (!canaryStages) {
+    if (!canaryStageInstance) {
       throw new IllegalStateException("No upstream canary stage found (stageId: ${stage.id}, executionId: ${stage.execution.id})")
     }
 
-    def canary = canaryStages.first()
-    def cancelResult = ((CancellableStage) canary.stageBuilder)?.cancel(canary.stage)
+    CanaryStage canaryStageBuilder = stageDefinitionBuilders.find {
+      it instanceof CanaryStage
+    }
```

Comment:
Can't you just inject the single `CanaryStage` rather than the whole list? Or are you adding variants for Kayenta?

Reply:
Had no good reason. Fixed.